server/src/main/java/org/elasticsearch/index/get/GetResult.java

```diff
@@ -224,19 +230,24 @@ public String sourceAsString() {
     }
 
     public Map<String, DocumentField> getFields() {
+        Map<String, DocumentField> fields = emptyMap();
+        fields.putAll(metaFields);
+        fields.putAll(otherFields);
+        // TODO: question to reviewers - do you think it makes sense to cache this merged `fields` map?
```

Comment:
I wouldn't worry about caching for now. Separately from this PR we should look at changing all the consumers of this method.

Reply:
ok